stackrox/stackrox
# ROX-12975: Extend Enricher to accept NodeInventory as a second parameter

## Description

This PR was created as a split from #4430 in order to make the change easier to review.
This PR is meant to replace #3940.

Here, we change the call to EnrichNode, so that it accepts two parameters: Node and NodeInventory. This allows us to keep Node and NodeInventory proto messages separate.

⚠️ This implementation is prone to a race condition between the central pipelines: Node and NodeInventory. This issue will be resolved in a follow-up ([ROX-14484](https://issues.redhat.com/browse/ROX-14484)).

## Checklist
- [x] Unit test and regression tests added (to some degree...)

### Not apply

## Testing Performed

- CI
- Deploy locally on Colima, enable FF and check logs
- Deploy locally on Colima with https://github.com/stackrox/scanner/pull/1059 and https://github.com/stackrox/stackrox/pull/4484 and see this in the UI

![Screenshot 2023-01-18 at 18 05 45](https://user-images.githubusercontent.com/114479/213247999-c1851d1b-aaa4-4af3-959e-cc2525bf4b9f.png)

- Deploy on OpenShift 4.11 (with Scanner PR https://github.com/stackrox/scanner/pull/1065) and see this in the UI

![Screenshot 2023-01-27 at 15 11 00](https://user-images.githubusercontent.com/114479/215121830-370e66e6-2be6-4b98-84fd-aa6246de7b61.png)
